src/controller/scan/base_controller.go

```diff
@@ -719,6 +735,18 @@ func (bc *basicController) GetScanLog(ctx context.Context, uuid string) ([]byte,
 	return b.Bytes(), nil
 }
 
+func taskForArtifacts(task *task.Task, artifactIDs []int64) bool {
```

Comment:
It's better to use a map for the second argument so that we don't need to loop the slice again and again.